public/tutorials/using-infura-custom-provider.md

```diff
@@ -1,136 +1,152 @@
+The next step is to edit your `truffle.js` file to use `HDWalletProvider` and provide all the necessary configuration for deploying to Ropsten.
 
-```javascript
-// Get our mnemonic and create an hdwallet
-var mnemonic = "couch solve unique spirit wine fine occur rhythm foot feature glory away";
-var hdwallet = hdkey.fromMasterSeed(bip39.mnemonicToSeed(mnemonic));
-```
+1. First, define the provider in the configuration. Add this line at the top of your `truffle.js` file:
 
-From here we can use the hdwallet to derive the first account. We do so using something called an "hd path", which is a way of telling the complex mathematics what we're attempting to derive. The most important bit is when we add "0" on the end of the hd path. This means we want to find the first account that we can derive from this hd wallet. It's zero based, so if we wanted the second address we'd derive `wallet_hdpath + "1"`, and for the third address `wallet_hdpath + "2"`, etc.
+   ```javascript
+   var HDWalletProvider = require("truffle-hdwallet-provider");
+   ```
 
-```javascript
-// Get the first account using the standard hd path.
-var wallet_hdpath = "m/44'/60'/0'/0/";
-var wallet = hdwallet.derivePath(wallet_hdpath + "0").getWallet();
-var address = "0x" + wallet.getAddress().toString("hex");
-```
+2. Next, provide a reference to your mnemonic that generates your accounts.
 
-Next, we need to set up the Provider Engine, telling it that we'd like to use our wallet to sign transactions, and use the ropsten network on infura for everything else:
+   ```javascript
+   var mnemonic = "orange apple banana ... ";
+   ```
 
-```javascript
-var providerUrl = "https://testnet.infura.io";
-var engine = new ProviderEngine();
-// filters
-engine.addProvider(new FilterSubprovider());
+   <p class="alert alert-danger">
+   <strong>Warning</strong>: In production, we highly recommend storing the mnemonic in another (secret) file, to reduce the risk of the mnemonic becoming known. If someone knows your mnemonic, they have all of your addresses and private keys!
+   </p>
 
-engine.addProvider(new WalletSubprovider(wallet, {}));
-engine.addProvider(new Web3Subprovider(new Web3.providers.HttpProvider(providerUrl)));
-engine.start(); // Required by the provider engine.
-```
+3. Add a Ropsten network definition:
 
-Finally, we want to export our Truffle configuration. Here we use this wallet only when we want to deploy to the ropsten network:
-
-```javascript
-module.exports = {
-  networks: {
-    "ropsten": {
-      network_id: 3,    // Official ropsten network id
-      provider: engine, // Use our custom provider
-      from: address     // Use the address we derived
-    }
-  },
-  rpc: {
-    // Use the default host and port when not using ropsten
-    host: "localhost",
-    port: 8545
-  }
-};
-```
+   ```javascript
+   module.exports = {
+     networks: {
+       ropsten: {
+         provider: new HDWalletProvider(mnemonic, "https://ropsten.infura.io/<INFURA_Access_Token>"),
```

Comment:
One thing I forgot to mention: We should stick this within a function. aka:

```
provider: function() {
  return new HDWalletProvider(mnemonic, "https://ropsten.infura.io/<INFURA_Access_Token>")
}
```

This will ensure the provider doesn't get initialized until it's needed, especially if they connect to multiple networks as you suggested below.